src/Carnac/ConcurrencyService.cs

```diff
@@ -0,0 +1,27 @@
+using System.Reactive.Concurrency;
+using System.Threading;
+
+namespace Carnac
+{
+    public class ConcurrencyService : IConcurrencyService
+    {
+        readonly IScheduler uiScheduler;
+        readonly IScheduler defaultScheduler;
+
+        public ConcurrencyService()
+        {
+            uiScheduler = new SynchronizationContextScheduler(SynchronizationContext.Current);
+            defaultScheduler = Scheduler.Default;
+        }
+
+        public IScheduler UiScheduler
```

Comment:
I like `DispatcherScheduler` (Rx) or `MainThreadScheduler` (ReactiveUI) over this name